File: POO Youtube Python/Gafanhoto.py
from Pessoa import Pessoa


class Gafanhoto(Pessoa):
    # Atributos
    __totassis = 0

    # Construtor
    def __init__(self, Nome: str, Idade: int, Sexo: str, Login: str):
        super().__init__(Nome, Idade, Sexo)
        self.login = Login

    # Metodos publicos
    def viumaisum(self):
        self.__totassis += 1
        self.ganharexp(100)

    # Metodos especiais
    def __repr__(self):
        return f'{self.nome}, {self.idade}, {self.sexo}, {self.login}, {self.__totassis}, {self.exp}'

File: Poo Youtube Kotlin/src/main/kotlin/Gafanhoto.kt
class Gafanhoto(Login: String, Nome: String, Idade: Int, Sexo: String): Pessoa(Nome, Idade, Sexo) {
    //Atributos
    var login = Login
        protected set
    var totasst: Int = 0
        protected set
    var likes: Int = 0
        protected set

    //Metodos Publicos
    fun viuMaisUm(){
        this.totasst += 1
        ganharExperiencia(100)
    }
    fun detalhesgaf(){
        detalhespess()
        println("Login: $login, Total assistido: $totasst")
    }
}